modules/shared_data/src/main/scala/org/elpaca_metagraph/shared_data/types/ConstellationBlockExplorer.scala

```diff
@@ -0,0 +1,26 @@
+package org.elpaca_metagraph.shared_data.types
+
+import derevo.circe.magnolia.{decoder, encoder}
+import derevo.derive
+import org.tessellation.schema.address.Address
+
+object ConstellationBlockExplorer {
+
+  @derive(encoder, decoder)
+  case class BlockExplorerTransaction(
+    hash       : String,
+    source     : Address,
+    destination: Address,
+    amount     : Long,
+    timestamp  : String
+  ) {
+    override def equals(obj: Any): Boolean = obj match {
+      case that: BlockExplorerTransaction => this.hash == that.hash
+      case _ => false
+    }
+
+    override def hashCode(): Int = hash.hashCode()
```

Comment:
I understand this PR has already been merged but this code raises a huge red flag for me because

- We're using Java overrides rather than Scala code
- We're changing the meaning of equivalence for this class.

Assuming this is really what we want -- and those situations are rare -- I feel there should at least be a comment explaining why we're doing this and so that other readers know that the standard behavior for instances of this case class does not apply

Reply:
I support that. For comparison, you can use `Eq` from cats, so you don't need to bother with native code